modules/nf-lineage/src/main/nextflow/lineage/DefaultLinStore.groovy

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2013-2025, Seqera Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package nextflow.lineage
+
+import java.nio.file.FileVisitResult
+import java.nio.file.FileVisitor
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.attribute.BasicFileAttributes
+
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+import nextflow.lineage.serde.LinEncoder
+import nextflow.lineage.serde.LinSerializable
+import nextflow.lineage.config.LineageConfig
+import nextflow.exception.AbortOperationException
+import nextflow.file.FileHelper
+import nextflow.util.TestOnly
+
+/**
+ * Default Implementation for the a lineage store.
+ *
+ * @author Paolo Di Tommaso <paolo.ditommaso@gmail.com>
+ */
+@Slf4j
+@CompileStatic
+class DefaultLinStore implements LinStore {
+
+    private static String HISTORY_FILE_NAME = ".history"
+    private static final String METADATA_FILE = '.data.json'
+    private static final String METADATA_PATH = '.meta'
```

Comment:
@jorgee what is the rationale for the `.meta` subfolder? It looks like the only thing that is created in the `lineage` folder. Do you intend to store other things under `lineage` as well?

Reply:
@bentsherman, it was storing the output data and the metadata in the first implementation, but currently this sub-folder is not needed.

Reply:
Okay, not urgent but something to consider in the final cleanup before 25.04